src/Features/Core/Portable/DesignerAttribute/AbstractDesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -22,154 +19,51 @@ namespace Microsoft.CodeAnalysis.DesignerAttribute
     [ExportWorkspaceService(typeof(IDesignerAttributeDiscoveryService)), Shared]
     internal sealed partial class DesignerAttributeDiscoveryService : IDesignerAttributeDiscoveryService
     {
-        /// <summary>
-        /// Protects mutable state in this type.
-        /// </summary>
-        private readonly SemaphoreSlim _gate = new SemaphoreSlim(initialCount: 1);
-
-        /// <summary>
-        /// Keep track of the last information we reported.  We will avoid notifying the host if we recompute and these
-        /// don't change.
-        /// </summary>
-        private readonly ConcurrentDictionary<DocumentId, (string? category, VersionStamp projectVersion)> _documentToLastReportedInformation = new();
```

Comment:
this is the remote side.  it became completely stateless (which is good).  there was a lot of complexity about the remote side trying to keep track of what hte host side might now and preventing updates.  this was error prone and doesn't work in the IASyncEnumerable world.  e.g. in remote IAE, jsut because you yielded the data, you dont' know that hte host has actually gotten it.  

this moved to a cleaner model where the remote side knows nothing, and the host side keeps track.